r/vignettes/dataset.Rmd

```diff
@@ -313,27 +330,29 @@ instead of a file path, or simply concatenate them like `big_dataset <- c(ds1, d
 
 As you can see, querying a large dataset can be made quite fast by storage in an
 efficient binary columnar format like Parquet or Feather and partitioning based on
-columns commonly used for filtering. However, we don't always get our data delivered
-to us that way. Sometimes we start with one giant CSV. Our first step in analyzing data
+columns commonly used for filtering. However, data isn't always stored that way.
+Sometimes you might start with one giant CSV. The first step in analyzing data 
 is cleaning is up and reshaping it into a more usable form.
 
-The `write_dataset()` function allows you to take a Dataset or other tabular data object---an Arrow `Table` or `RecordBatch`, or an R `data.frame`---and write it to a different file format, partitioned into multiple files.
+The `write_dataset()` function allows you to take a Dataset or another tabular 
+data object - an Arrow Table or RecordBatch, or an R data frame - and write
```

Comment:
```suggestion
data object - an Arrow Table or RecordBatch, or an R data.frame - and write
```

This is another we should probably standardize on. This isn't perfect, but as a metric: I see 47 results for `data.frame` in arrow/r/man but only 5 results for `data frame`

Reply:
Here's how I think about it when writing: "`data.frame`" is a proper noun, meaning literally that class of R object, while "data frame" is the more conceptual noun. In some cases, one or the other is obviously correct; in others, either might work.

Reply:
How about in this example here @nealrichardson ? I was thinking `data frame` as we're referring to both `data.frame` and `tibble::tibble` objects but then they're both of or inherit from class `data.frame ` so wasn't sure what makes most sense.

Reply:
This is an example of both I found and was trying to emulate: https://r4ds.had.co.nz/tibbles.html

Reply:
I know 😞  I think I'd go with "data frame" so as to (in theory) avoid the "you said data.frame but it's a tibble" though my reasoning for picking one or the other is more about what's in my head when I'm writing, and you're right, the distinction is not perfect in the real world, nor may the reader share my distinction.

Reply:
I'm leaning towards this interpretation too, so have added guidance in `STYLE.md`.

Reply:
more em-dashes

```suggestion
data object—an Arrow Table or RecordBatch, or an R data frame—and write
```